server/auth/types/openid/openid_auth.ts

```diff
@@ -124,6 +128,22 @@ export class OpenIdAuthentication extends AuthenticationType {
     }
   }
 
+  private generateNextUrl(request: OpenSearchDashboardsRequest): string {
+    const path =
+      this.coreSetup.http.basePath.serverBasePath +
+      (request.url.pathname || '/app/opensearch-dashboards');
+    return escape(path);
+  }
+
+  private redirectOIDCCapture = (request: OpenSearchDashboardsRequest, toolkit: AuthToolkit) => {
+    const nextUrl = this.generateNextUrl(request);
+    const clearOldVersionCookie = clearOldVersionCookieValue(this.config);
```

Comment:
I was taking a look of the `clearOldVersionCookieValue`, and it seems like the goal is to clear a specific cookie named "security_authentication" by expiring it immediately. So I just wanna double confirm that we are doing this on purpose is because we would like to clean the auth info that stored in the "security_authentication" cookies?

Reply:
If its re-directing to the IdP's login its because the current cookie is not valid. This explicitly clears the bad cookie.

I'm not very familiar with this function so I'll look into the history of it.

Reply:
It was introduced in this PR: https://github.com/opensearch-project/security-dashboards-plugin/pull/620